### ISSUE TITLE
[BUG] SearchServiceCollection.CreateOrUpdateAsync throws ObjectDisposedException

### ISSUE THREAD

**Issue by tbajkacz** [Azure/azure-sdk-for-net#39523]
### Library name and version

Azure.ResourceManager.Search_1.2.0-beta.1

### Describe the bug

**SearchServiceCollection.CreateOrUpdateAsync** throws **ObjectDisposedException**. 

The service gets created regardless of the exception. Problem occurs both for WaitUntil.Started and WaitUntil.Completed.

### Expected behavior

No exception should be thrown when successfully creating or updating a search service.

### Actual behavior

An exception is thrown when attempting to create a search service. The service gets created regardless of the exception.

### Reproduction Steps

```csharp
var client = new ArmClient(new DefaultAzureCredential());
var subscriptions = client.GetSubscriptions();
var subscription = await subscriptions.GetAsync(_configuration.SubscriptionId);
var resourceGroup = await subscription.Value.GetResourceGroupAsync(_configuration.ResourceGroupName);
var searchServiceCollection = resourceGroup.Value.GetSearchServices();

// exception is thrown for both WaitUntil.Started and WaitUntil.Completed
await searchServiceCollection.CreateOrUpdateAsync(WaitUntil.Started, @params.NewServiceName, new(AzureLocation.UKSouth)
{
   SkuName = SearchSkuName.Basic,
   ReplicaCount = 1,
   PartitionCount = 1,
   HostingMode = SearchServiceHostingMode.Default,
   PublicNetworkAccess = SearchServicePublicNetworkAccess.Enabled
});
```

### Environment

MacOS, .NET 7.0.100 Rider

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by mcgallan** [comment]
@tbajkacz , The issue you mentioned has been fixed in version 1.3.0-beta.1. Thank you for your feedback!
